python/cudf/cudf/core/dataframe.py

```diff
@@ -2627,7 +2627,64 @@ def reindex(
 
         return DataFrame(cols, idx)
 
-    def set_index(self, index, drop=True):
+    def _set_index(
+        self,
+        index,
+        drop=True,
+        append=False,
+        inplace=False,
+        verify_integrity=False,
+    ):
+        """Helper for `.set_index`
+
+        Parameters
+        ----------
+        index : Index
+            The new index to set.
+        drop : boolean
+            Whether to drop corresponding column for str index argument
+        append : boolean
+            Append current to new index to form a multiindex
+        inplace : boolean
+            Modify the DataFrame in place (do not create a new object)
+        verify_integrity : boolean
+            Check for duplicates in the new index
+        """
+        assert isinstance(index, Index)
```

Comment:
Would suggest we raise an Informative error here.

Reply:
If we want to validate something don't use assertions, they can be removed with `python -o`, explicitly check and throw instead.